Composer/packages/client/src/recoilModel/dispatchers/notification.ts

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { CallbackInterface, useRecoilCallback } from 'recoil';
+import { v4 as uuid } from 'uuid';
+
+import { notificationsState } from '../atoms/appState';
+import { ICardProps } from '../../components/NotificationCard';
+
+export const addNotificationInternal = ({ set }: CallbackInterface, notification: ICardProps) => {
+  const id = uuid(6);
+  set(notificationsState, (notifications) => [...notifications, { id, cardProps: notification }]);
+  return id;
```

Comment:
This is an anti-pattern. Instead, extract a helper method to create a notification and then invoke this dispatcher with the created notification.